hybrid/flow.py

```diff
@@ -587,13 +588,107 @@ def next(self, states):
         return states[min_idx]
 
 
-class Loop(Runnable):
+class TrackMin(Runnable, traits.SISO):
+    """Tracks (and keeps record of) the "best" (or minimal) :class:`State` seen,
+    according to a metric defined with a `key` function.
+
+    Args:
+        key (callable/str, optional, default=None):
+            Best state is judged according to a metric defined with a `key`.
+            `key` can be a `callable` with a signature::
+
+                key :: (State s, Ord k) => s -> k
+
+            or a string holding a key name/path to be extracted from the input
+            state with `operator.attrgetter` method.
+
+            By default, `key == operator.attrgetter('samples.first.energy')`,
+            thus favoring states containing a sample with the minimal energy.
```

Comment:
Above you use :class:`State` but here you switched to state, and below

Reply:
We don't have a clear standard on this (that I know of) - topic for a discussion. I'll update "state" to "State".